okr/scrapers/podcasts/__init__.py

```diff
@@ -1,592 +1,670 @@
+                spotify_episode_ids_search, ep_metas_public
+            ):
+                if ep_meta_public:
+                    spotify_episode_id_by_name[ep_meta_public["name"]] = episode_id
+        # Leave Spotify ID empty of no matching ID was found
+        else:
+            spotify_episode_id_by_name = {}
+
+        # Read data from RSS feed
+        d = feed.parse(podcast.feed_url)
+        for entry in d.entries:
+            spotify_id = spotify_episode_id_by_name.get(entry.title)
+
+            media_url = entry.enclosures[0].href
+            zmdb_id = int(media_url.split("/")[-2])
+            t = dt.datetime.strptime(entry.itunes_duration, "%H:%M:%S")
+            duration = dt.timedelta(hours=t.hour, minutes=t.minute, seconds=t.second)
+            publication_date_time = dt.datetime(
+                *entry.published_parsed[:6], tzinfo=pytz.UTC
+            )
+            defaults = {
+                "podcast": podcast,
+                "title": entry.title,
+                "description": entry.description,
+                "publication_date_time": publication_date_time,
+                "media": media_url,
+                "spotify_id": spotify_id,
+                "duration": duration,
+            }
+
+            try:
+                obj, created = PodcastEpisode.objects.update_or_create(
+                    zmdb_id=zmdb_id,
+                    defaults=defaults,
+                )
+            except IntegrityError as e:
+                capture_exception(e)
+                print(
+                    f"Data for {entry.title} failed integrity check:",
+                    defaults,
+                    sep="\n",
+                )
+
+
+def scrape_spotify_api(
+    *, start_date: Optional[dt.date] = None, podcast_filter: Optional[Q] = None
+):
+    """Read and process data from Spotify API.
+
+    This method supplies two kinds of Spotify-related data:
+
+    a) Data about the podcast itself, such as the current number of the podcast's
+    followers
+
+    b) Client side usage data for each podcast episode. This includes starts, streams
+    (minimum listening duration of 1 minute), or listeners (number of accounts).
+
+    Args:
+        start_date (dt.date, optional): Earliest date to request data for. Defaults to
+          None.
+        podcast_filter (Q, optional): Filter for a subset of all Podcast objects.
+          Defaults to None.
+    """
+    today = local_today()
+    yesterday = local_yesterday()
+
+    if start_date is None:
+        start_date = yesterday - dt.timedelta(days=5)
+
+    podcasts = Podcast.objects.all()
+
+    if podcast_filter:
+        podcasts = podcasts.filter(podcast_filter)
+
+    for podcast in podcasts:
+        print("Scraping spotify API for", podcast)
+
+        if not podcast.spotify_id:
+            print("No Spotify ID for", podcast)
+            continue
+
+        # Retrieve follower and listener data for podcast (ignores start_date)

```

Comment:
Only follower data ignores `start_date`. Reason is that the API only supports querying the current follower number, no historic data is provided.